src/documentation/pipelines/create-your-pipeline.md

```diff
@@ -5,4 +5,153 @@ permalink: /documentation/pipelines/create-your-pipeline/
+*   Apply **transforms** to each `PCollection`. Transforms can change, filter, group, analyze, or otherwise process the elements in a `PCollection`. Each transform creates a new output `PCollection`, to which you can apply additional transforms until processing is complete.
+*   **Write** or otherwise output the final, transformed `PCollection`s.
+*   **Run** the pipeline.
+
+## Creating Your Pipeline Object
+
+A Beam program often starts by creating a `Pipeline` object.
+
+In the Beam SDKs, each pipeline is represented by an explicit object of type `Pipeline`. Each `Pipeline` object is an independent entity that encapsulates both the data the pipeline operates over and the transforms that get applied to that data.
+
+To create a pipeline, declare a `Pipeline` object, and pass it some configuration options, which are explained in a section below. You pass the configuration options by creating an object of type `PipelineOptions`, which you can build by using the static method `PipelineOptionsFactory.create()`.
+
+```java
+// Start by defining the options for the pipeline.
+PipelineOptions options = PipelineOptionsFactory.create();
+
+// Then create the pipeline.
+Pipeline p = Pipeline.create(options);
+```
+
+### Configuring Pipeline Options
+
+Use the pipeline options to configure different aspects of your pipeline, such as the pipeline runner that will execute your pipeline and any runner-specific configuration required by the chosen runner.
+
+For example, if you're using the Google Cloud Dataflow service to run your pipeline, your pipeline options will include information about your Cloud Platform project required by the Cloud Dataflow service, such as your project ID and Cloud Storage staging locations. The pipeline options also let you control how many workers the Dataflow service should assign to your pipeline job, and where to direct your pipeline job's status messages.
+
+A key property in the pipeline options that determines where your pipeline executes is the pipeline runner. The pipeline runner property also specifies whether your pipeline's execution is to be asynchronous or blocking.
+
+When you run the pipeline on a runner of your choice, a copy of the PipelineOptions will be available to your code. For example, you can read PipelineOptions from a DoFn's Context.
+
+> **Note:** Access `PipelineOptions` inside any [ParDo](/documentation/programming-guide/#transforms-pardo)'s `DoFn` instance by using the method `ProcessContext.getPipelineOptions`.
+
+#### Setting PipelineOptions from Command-Line Arguments
+
+While you can configure your pipeline by creating a `PipelineOptions` object and setting the fields directly, the Beam SDKs include a command-line parser that you can use to set fields in `PipelineOptions` using command-line arguments.
+
+To read options from the command-line, construct your `PipelineOptions` object using the method `PipelineOptionsFactory.fromArgs`, as in the following example code:
```

Comment:
the formatting of fromArgs is going to be language-dependent. remove from prose?